### PR TITLE
fix: 🐛 Different default error msg for inputs like radio, check

### PR DESCRIPTION
**Pull request by dBucik**
Provide more meaningful default error messages for inputs that do not have format, e.g. radio, checkbox, select, embedded_gr_app.

### CODE REVIEW

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/PerunFormItemValidatorImpl.java [hunk 1]
```diff
@@ -41,7 +41,21 @@ public String getErrorMsgOrDefault(T item) {
 		}
 
 		if (errorText == null || errorText.isEmpty()) {
-			return translation.incorrectFormat();
+			switch (item.getItemData().getFormItem().getType()) {
```
Comment (zlamalp) [inline perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/PerunFormItemValidatorImpl.java:44]: I'm not sure we need this for all of the form item types. This message is reserved for the cases, when regex validation of the form item fails. If you expect non-empty value (selected at least one option), it should be done through required flag of the form item. Emptiness is checked before the regex and should display correct messages for the form items (about selecting the option).

I found only one case (but maybe there are more) when we resort to this error message instead and its checkbox item with the single checkbox option. In such a case we prefer to display this customized error message (like we expect you to say "you need to approve the AUP" or whatever).

File: perun-wui-registrar/src/main/resources/cz/metacentrum/perun/wui/registrar/client/resources/PerunRegistrarTranslation_cs.properties [hunk 2]
```diff
@@ -1,177 +1,181 @@
 # // -------------- APP NAME------------------------ //
 
 registrarAppName=Registrace
-pleaseVerifyCaptcha=Pro pokračování prosím proveďte potvrzení níže
-captchaFailed=Ověření selhalo. Prosím, proveďte potvrzení znovu.
+pleaseVerifyCaptcha=Pro pokra\u00C4\u008Dov\u00C3\u00A1n\u00C3\u00AD pros\u00C3\u00ADm prove\u00C4\u008Fte potvrzen\u00C3\u00AD n\u00C3\u00AD\u00C5\u00BEe
```
Comment (zlamalp) [inline perun-wui-registrar/src/main/resources/cz/metacentrum/perun/wui/registrar/client/resources/PerunRegistrarTranslation_cs.properties:4]: Our source files should be in UTF-8 so there is no need to convert the messages. I think you should update your IDE settings to respect UTF-8 for this file (IDE usually does this transparent translation and think these files are ASCII only).

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/PerunFormItemValidatorImpl.java (hunk 1 above)
Comment (dBucik) [reply perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/PerunFormItemValidatorImpl.java:44]: You are right, I have not noticed that. So In the amended commit, I will add just an alternative translation for that edge-case scenario :).

File: perun-wui-registrar/src/main/resources/cz/metacentrum/perun/wui/registrar/client/resources/PerunRegistrarTranslation_cs.properties (hunk 2 above)
Comment (dBucik) [reply perun-wui-registrar/src/main/resources/cz/metacentrum/perun/wui/registrar/client/resources/PerunRegistrarTranslation_cs.properties:4]: Thanks, did not notice. For some reason when I tried to switch to UTF-8, it re-encoded the whole file, so I left it as it was. Seems it did the re-encoding to ISOwhatever on the fly without marking changes.

File: perun-wui-registrar/src/main/resources/cz/metacentrum/perun/wui/registrar/client/resources/PerunRegistrarTranslation_cs.properties [hunk 3]
```diff
@@ -55,6 +55,7 @@ duplicateKeys=Každý klíč musí být unikátní!
 cantBeEmpty=Položka <b>nesmí být prázdná!</b>
 cantBeEmptySelect=Musíte vybrat jednu z možností!
 cantBeEmptyCheckBox=Musíte vybrat alespoň jednu možnost!
+cantBeEmptySingleCheckBox=Tuhle položku musíte označiť!
```
Comment (zlamalp) [inline perun-wui-registrar/src/main/resources/cz/metacentrum/perun/wui/registrar/client/resources/PerunRegistrarTranslation_cs.properties:58]: `označit` not `označiť`

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/CheckboxValidator.java [hunk 4]
```diff
@@ -24,7 +24,7 @@ public boolean validateLocal(Checkbox checkbox) {
 			} else {
 				// single checkbox - prefer own error message
 				setResult(Result.EMPTY);
-				checkbox.setRawStatus(getErrorMsgOrDefault(checkbox), ValidationState.ERROR);
+				checkbox.setRawStatus(getTransl().cantBeEmptySingleCheckBox(), ValidationState.ERROR);
```
Comment (zlamalp) [inline perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/CheckboxValidator.java:27]: Now you prevent the possibility to display custom error message (eg. "You must agree to the AUP"). You should probably check if returned error message is the same as the default message and in that case replace it with your message. I'm not sure if there is getErrorMsg() without the default, then you could just check if it returns empty/null message and use your message instead.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/CheckboxValidator.java (hunk 4 above)
Comment (dBucik) [reply perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/widgets/items/validators/CheckboxValidator.java]: Should now be fixed as suggested.

**Review by zlamalp — APPROVED** [review]

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**